xarray/conventions.py

```diff
@@ -502,6 +507,20 @@ def stackable(dim):
                     new_vars[k].encoding["coordinates"] = coord_str
                     del var_attrs["coordinates"]
                     coord_names.update(var_coord_names)
+            if "bounds" in var_attrs:
+                bounds_str = var_attrs["bounds"]
+                var_bounds_names = [bounds_str]
+                if all(k in variables for k in var_bounds_names):
+                    new_vars[k].encoding["bounds"] = bounds_str
+                    coord_names.update(var_bounds_names)
+                del var_attrs["bounds"]
+            if "grid_mapping" in var_attrs:
+                proj_str = var_attrs["grid_mapping"]
+                var_proj_names = proj_str.split()
+                if all(k in variables for k in var_proj_names):
```

Comment:
if the attribute mentions variables that doesn't exist, shall we raise a warning about the nonexistent variables and convert the others to coords?

Reply:
Added the warning.